src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzerTests.cs

```diff
@@ -1080,6 +1081,164 @@ public void M1()
+        [|UnsupportedOnBrowserType.SupportedOnTvos4()|];
+        UnsupportedOnBrowserType.SupportedOnBrowser(); // child support ignored (should not extend)
+        UnsupportedOnBrowserType.UnsupportedOnBrowser();
+    }
+}
+
+[SupportedOSPlatform(""browser"")]
+static class Program2
+{
+    public static void Main()
+    {
+        [|UnsupportedOnBrowserType.SupportedOnWindowsIosTvos()|]; // 4 diagnostics expected
+        [|UnsupportedOnBrowserType.SupportedOnTvos4()|];
+        [|UnsupportedOnBrowserType.SupportedOnBrowser()|];    // only warns for parent attribute
+        [|UnsupportedOnBrowserType.UnsupportedOnBrowser()|];
+    }
+}
+
+[UnsupportedOSPlatform(""browser"")]
+[SupportedOSPlatform(""windows"")]
+static class Program3
+{
+    public static void Main()
+    {
+        UnsupportedOnBrowserType.SupportedOnWindowsIosTvos(); // No diagnostics expected
+        [|UnsupportedOnBrowserType.SupportedOnTvos4()|];
+        UnsupportedOnBrowserType.SupportedOnBrowser();
+        UnsupportedOnBrowserType.UnsupportedOnBrowser();
+    }
+}
+
+[UnsupportedOSPlatform(""browser"")]
+class UnsupportedOnBrowserType
+{
+    [SupportedOSPlatform(""windows"")]
+    [SupportedOSPlatform(""ios"")]
+    [SupportedOSPlatform(""tvos"")] 
+    public static void SupportedOnWindowsIosTvos() {}
+    [SupportedOSPlatform(""browser2.0"")] 
+    public static void SupportedOnBrowser() {}
+    [UnsupportedOSPlatform(""browser1.0"")] 
+    public static void UnsupportedOnBrowser() {}
+    [SupportedOSPlatform(""tvos4.0"")]    
+    public static void SupportedOnTvos4() {}
+}
+" + MockAttributesCsSource;
+            await VerifyAnalyzerAsyncCs(source, VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.SupportedOsRule).WithLocation(9, 9)
+                    .WithMessage("'UnsupportedOnBrowserType.SupportedOnWindowsIosTvos()' is supported on 'ios'"),
+                    VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.UnsupportedOsRule).WithLocation(9, 9)
+                    .WithMessage("'UnsupportedOnBrowserType.SupportedOnWindowsIosTvos()' is supported on 'tvos'"),
+                    VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.UnsupportedOsRule).WithLocation(21, 9)
+                    .WithMessage("'UnsupportedOnBrowserType.SupportedOnWindowsIosTvos()' is unsupported on 'browser'"),
+                    VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.SupportedOsRule).WithLocation(21, 9)
+                    .WithMessage("'UnsupportedOnBrowserType.SupportedOnWindowsIosTvos()' is supported on 'ios'"),
+                    VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.UnsupportedOsRule).WithLocation(21, 9)
+                    .WithMessage("'UnsupportedOnBrowserType.SupportedOnWindowsIosTvos()' is supported on 'tvos'"),
+                    VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.UnsupportedOsRule).WithLocation(22, 9)
+                    .WithMessage("'UnsupportedOnBrowserType.SupportedOnTvos4()' is unsupported on 'browser'"));
```

Comment:
```suggestion
            await VerifyAnalyzerAsyncCs(source, 
            	VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.SupportedOsRule)
            		.WithLocation(9, 9)
                    .WithMessage("'UnsupportedOnBrowserType.SupportedOnWindowsIosTvos()' is supported on 'ios'"),
                VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.UnsupportedOsRule)
                	.WithLocation(9, 9)
                    .WithMessage("'UnsupportedOnBrowserType.SupportedOnWindowsIosTvos()' is supported on 'tvos'"),
                VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.UnsupportedOsRule)
                	.WithLocation(21, 9)
                    .WithMessage("'UnsupportedOnBrowserType.SupportedOnWindowsIosTvos()' is unsupported on 'browser'"),
                VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.SupportedOsRule)
                	.WithLocation(21, 9)
                    .WithMessage("'UnsupportedOnBrowserType.SupportedOnWindowsIosTvos()' is supported on 'ios'"),
                VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.UnsupportedOsRule)
                	.WithLocation(21, 9)
                    .WithMessage("'UnsupportedOnBrowserType.SupportedOnWindowsIosTvos()' is supported on 'tvos'"),
                VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.UnsupportedOsRule)
                	.WithLocation(22, 9)
                    .WithMessage("'UnsupportedOnBrowserType.SupportedOnTvos4()' is unsupported on 'browser'"));
```

Reply:
All other tests written in this way, as this is a test code I didn't want to break it down so many rows. It is broken down just to make sure it was visible in github without scrolling